examples/cpp/dds/CustomPayloadPoolExample/CustomPayloadPoolDataSubscriber.h

```diff
@@ -0,0 +1,78 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file CustomPayloadPoolDataSubscriber.h
+ *
+ */
+
+#ifndef CUSTOM_PAYLOAD_POOL_DATA_SUBSCRIBER_H_
+#define CUSTOM_PAYLOAD_POOL_DATA_SUBSCRIBER_H_
+
+#include "CustomPayloadPoolDataPubSubTypes.h"
+#include "CustomPayloadPool.hpp"
+
+#include <fastdds/dds/domain/DomainParticipant.hpp>
+#include <fastdds/dds/subscriber/DataReaderListener.hpp>
+#include <fastrtps/subscriber/SampleInfo.h>
+#include <fastdds/dds/core/status/SubscriptionMatchedStatus.hpp>
+
+class CustomPayloadPoolDataSubscriber : public eprosima::fastdds::dds::DataReaderListener
+{
+public:
+
+    CustomPayloadPoolDataSubscriber(
+            std::shared_ptr<CustomPayloadPool> payload_pool);
+
+    virtual ~CustomPayloadPoolDataSubscriber();
+
+    //!Initialize the subscriber
+    bool init();
+
+    //!RUN the subscriber
+    void run();
```

Comment:
I'd follow the same pattern as in the publisher, that is run(0) means forever. In that case we don't need this API
```suggestion
```